src/mds/query.py

```diff
@@ -102,14 +103,46 @@ def add_filter(query):
         ]
 
 
+@mod.get("/metadata/{guid:path}/aliases")
+async def get_metadata_aliases(
+    guid: str,
+) -> JSONResponse:
+    """
+    Get the aliases for the provided GUID
+
+    Args:
+        guid (str): Metadata GUID
+    """
+    metadata_aliases = await MetadataAlias.query.where(
+        MetadataAlias.guid == guid
+    ).gino.all()
+
+    aliases = [metadata_alias.alias for metadata_alias in metadata_aliases]
+    return {"guid": guid, "aliases": sorted(aliases)}
+
+
 @mod.get("/metadata/{guid:path}")
 async def get_metadata(guid):
     """Get the metadata of the GUID."""
     metadata = await Metadata.get(guid)
-    if metadata:
-        return metadata.data
-    else:
-        raise HTTPException(HTTP_404_NOT_FOUND, f"Not found: {guid}")
+
+    if not metadata:
+        # check if it's an alias
+        alias = guid
+        metadata_alias = await MetadataAlias.query.where(
+            MetadataAlias.alias == alias
+        ).gino.first()
+
+        if not metadata_alias:
+            raise HTTPException(HTTP_404_NOT_FOUND, f"Not found: {guid}")
+
+        # get metadata for guid based on alias
+        metadata = await Metadata.get(metadata_alias.guid)
+
+        if not metadata:
+            raise HTTPException(HTTP_404_NOT_FOUND, f"Not found: {guid}")
```

Comment:
I'm wondering if this should be a 500 instead of a 404. It should not be possible to have an entry in `MetadataAlias` that has no corresponding entry in `Metadata`. Or if you want to keep the 404 maybe we can add a warning log?